packages/devtools_app/lib/src/screens/performance/panes/queued_microtasks/queued_microtasks_view.dart

```diff
@@ -0,0 +1,278 @@
+  }) : _controller = controller;
+
+  final QueuedMicrotasksController _controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<QueuedMicrotasksControllerStatus>(
+      valueListenable: _controller.status,
+      builder: (_, status, _) {
+        return RefreshButton(
+          iconOnly: true,
+          outlined: false,
+          onPressed: status == QueuedMicrotasksControllerStatus.refreshing
+              ? null
+              : _controller.refresh,
+          tooltip:
+              "Take a new snapshot of the selected isolate's microtask queue.",
+          gaScreen: gac.performance,
+          gaSelection: gac.PerformanceEvents.refreshQueuedMicrotasks.name,
+        );
+      },
+    );
+  }
+}
+
+class QueuedMicrotasksTabControls extends StatelessWidget {
+  const QueuedMicrotasksTabControls({
+    super.key,
+    required QueuedMicrotasksController controller,
+  }) : _controller = controller;
+
+  final QueuedMicrotasksController _controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [RefreshQueuedMicrotasksButton(controller: _controller)],
+    );
+  }
+}
+
+class RefreshQueuedMicrotasksInstructions extends StatelessWidget {
+  const RefreshQueuedMicrotasksInstructions({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: RichText(
+        text: TextSpan(
+          style: Theme.of(context).regularTextStyle,
+          children: [
+            const TextSpan(text: 'Click the refresh button '),
+            WidgetSpan(child: Icon(Icons.refresh, size: defaultIconSize)),
+            const TextSpan(
+              text:
+                  " to take a new snapshot of the selected isolate's "
+                  'microtask queue.',
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+// In the response returned by the VM Service, microtasks are sorted in
+// ascending order of when they will be dequeued, i.e. the microtask that will
+// run earliest is at index 0 of the returned list. We use those indices of the
+// returned list to sort the entries of the microtask selector, so that they
+// they also appear in ascending order of when they will be dequeued.
+typedef IndexedMicrotask = (int, Microtask);
+
+class _MicrotaskIdColumn extends ColumnData<IndexedMicrotask> {
+  _MicrotaskIdColumn()
+    : super.wide('Microtask ID', alignment: ColumnAlignment.center);
+
+  @override
+  int getValue(IndexedMicrotask indexedMicrotask) => indexedMicrotask.$1;
+
+  @override
+  String getDisplayValue(IndexedMicrotask indexedMicrotask) =>
+      indexedMicrotask.$2.id!.toString();
+}
+
+class QueuedMicrotaskSelector extends StatelessWidget {
+  const QueuedMicrotaskSelector({
+    super.key,
+    required List<IndexedMicrotask> indexedMicrotasks,
+    required void Function(Microtask?) setSelectedMicrotask,
```

Comment:
super nit: can we rename to `onMicrotaskSelected` to match the naming convention of the Table widget?